src/vs/workbench/services/search/common/search.ts

```diff
@@ -106,13 +106,19 @@ export interface ICommonQueryProps<U extends UriComponents> {
 	onlyFileScheme?: boolean;
 }
 
+export type FileQueryFilePatternType = 'fuzzy' | 'glob' | 'globCaseInsensitive';
+
 export interface IFileQueryProps<U extends UriComponents> extends ICommonQueryProps<U> {
 	type: QueryType.File;
 	filePattern?: string;
 
-	// when walking through the tree to find the result, don't use the filePattern to fuzzy match.
-	// Instead, should use glob matching.
-	shouldGlobMatchFilePattern?: boolean;
+	/**
+	 * Determines how the file pattern should be matched when walking through the tree to find the result:
+	 * - 'fuzzy': Use fuzzy matching (default)
+	 * - 'glob': Use glob pattern matching (case-sensitive)
+	 * - 'globCaseInsensitive': Use glob pattern matching (case-insensitive)
+	 */
+	filePatternType?: FileQueryFilePatternType;
```

Comment:
Maybe instead of a third type, we should consider keeping the old type and instead allowing the user to pass in the glob options